index.js

```diff
@@ -6,9 +6,22 @@ const { FileReadStream, FileWriteStream } = require('./streams.js')
 const mutexify = require('mutexify/promise')
 
 module.exports = class Localdrive {
-  constructor (root) {
+  constructor (root, opts = {}) {
     this.root = root
+    this.metadata = opts.metadata || new Map()
```

Comment:
Default should be null or noops

Reply:
What about `{}` so all the conditions like `if (this.metadata.get)` works?
Or always functions?